src/main/java/fi/aalto/cs/apluscourses/intellij/model/task/IntelliJComponentPresenterBase.java

```diff
@@ -46,6 +48,14 @@ public void highlight() {
 
   @RequiresEdt
   private void highlightInternal() {
+    var tutorialViewModel = PluginSettings.getInstance().getMainViewModel(project).tutorialViewModel.get();
+    var rootFrame = tutorialViewModel != null ? tutorialViewModel.getTutorialWindowFrame() : null;
+
+    // if we are focused on another window than the tutorial one, don't touch anything
+    if (JOptionPane.getRootFrame() != rootFrame) {
```

Comment:
To me, it seems that the condition could be replaced with
```java
JOptionPane.getRootFrame() != WindowManager.getInstance().getFrame(project)
```

Reply:
I wasn't aware of WindowManager